src/algokit_utils/_debugging.py

```diff
@@ -174,8 +175,18 @@ def _build_avm_sourcemap(  # noqa: PLR0913
     return AVMDebuggerSourceMapEntry(str(source_map_output_path), program_hash)
 
 
+@deprecated(
+    reason="Use latest version of `AlgoKit AVM Debugger` VSCode extension instead. "
+    "It will automatically manage your sourcemaps.",
+    version="3.0.0",
+)
 def persist_sourcemaps(
-    *, sources: list[PersistSourceMapInput], project_root: Path, client: "AlgodClient", with_sources: bool = True
+    *,
+    sources: list[PersistSourceMapInput],
+    project_root: Path,
+    client: "AlgodClient",
+    with_sources: bool = True,
+    persist_mappings: bool = False,
```

Comment:
Is there any need to have a flag that turns on this behaviour?

Reply:
Nope, good point - removed